selectors/selectors.md

```diff
@@ -117,12 +117,26 @@ type ExploreRange struct {
 ## Be careful when using ExploreRecursive with a large maxDepth parameter;
 ## it can easily cause very large traversals (especially if used in combination
 ## with selectors like ExploreAll inside the sequence).
+##
+## maxDepth is a union type -- it can have an integer value (key "depth") or
+## no value (key "none"). If maxDepth has no value it is up to the 
+## implementation library using selectors to identify an appropriate max depth
+## as neccesary so that recursion is not infinite
+
 type ExploreRecursive struct {
 	sequence Selector (rename ":>")
-	maxDepth Int (rename "d")
+	maxDepth RecursionLimit (rename "l")
```

Comment:
Could also change the field name to something else at this point, but that's just varnish.